mlflow/sklearn/utils.py

```diff
@@ -126,16 +133,38 @@ def _get_metrics_value_dict(metrics_list):
         try:
             metric_value = metric.function(**metric.arguments)
         except Exception as e:  # pylint: disable=broad-except
-            _log_warning_for_metrics(metric.name, metric.function, e)
+            _log_warning_for_metrics_or_artifacts(metric.name, metric.function, e, True)
```

Comment:
I think it would be cleaner / more extensible to break this into `_log_warning_for_metrics` and `_log_warning_for_artifacts`. In general, this pattern of defining specialized / branching logic based on a boolean parameter is brittle & prone to complexity later on (e.g., what happens if we add a third type of entity being logged & the desired warning message is substantially different?)

Reply:
For scalability, yes!